src/ProgramInitHelpers.cpp

```diff
@@ -0,0 +1,43 @@
+#include "defines.h" // IWYU pragma: keep
+#include "ProgramInitHelpers.h"
+#include <boost/locale.hpp>
+#include <boost/filesystem/path.hpp>
+#include <iostream>
+#include <cstdlib>
+#include <stdexcept>
+
+bool InitLocale()
+{
+    // Check and set locale (avoids errors caused by invalid locales later like #420)
+    try{
+        // Check for errors and use classic locale to avoid e.g. thousand separator in int2string conversions via streams
+#ifdef _WIN32
+        // On windows we want to enforce the encoding (mostly UTF8). Also using "" would use the default which uses "wrong" separators
+        std::locale::global(boost::locale::generator().generate("C"));
```

Comment:
There was a problem with Linux/OSX which I don't want to trigger again because I don't know if "C" will cause any change compared to "classic". In windows it seems to be the same. But "Never change a running system" applies here ;)